store/store.go

```diff
@@ -546,6 +559,110 @@ type requestOptions struct {
 	Data         []byte
 }
 
+type httpErrorResponse struct {
+	StatusCode int
+	URL        *url.URL
+	Method     string
+	Oops       string
+}
+
+func NewHttpErrorResponse(resp *http.Response) *httpErrorResponse {
+	return &httpErrorResponse{StatusCode: resp.StatusCode, URL: resp.Request.URL, Method: resp.Request.Method}
+}
+
+func (r httpErrorResponse) Error() string {
+	return fmt.Sprintf("HTTP request failed: '%v', method: %s, status: %d", r.URL, r.Method, r.StatusCode)
+}
+
+func (s *Store) doStoreRequest(client *http.Client, req *http.Request, user *auth.UserState, dataDecodeFunc func(*http.Response) error) error {
+	var err error
+	for refreshed := true; refreshed; {
+		refreshed = false
```

Comment:
Because it needs to run at least once. refreshed is immediately set to false but during the iteration it can be reset back to true.